runtime/src/comm/ugni/comm-ugni.c

```diff
@@ -5421,6 +5426,8 @@ void chpl_comm_execute_on(int locale, c_sublocid_t subloc,
            (int) locale, (int) subloc, (int) fid, arg, arg_size);
 
   if (locale == chpl_nodeID) {
+    assert(0); // locale model code should prevent this...
```

Comment:
I was duplicating some code in GASNet which uses asserts for this reason too. We could make them all error messages but there are several of them and I think assert is good enough. Did you see some particular advantage to doing halt-with error?